cmd/system-probe/modules/language_detection.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package modules
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	languageDetectionProto "github.com/DataDog/datadog-agent/pkg/proto/pbgo/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var LanguageDetectionModule = module.Factory{
+	Name:             config.LanguageDetectionModule,
+	ConfigNamespaces: []string{"language_detection"},
+	Fn: func(cfg *config.Config) (module.Module, error) {
+		return languageDetectionModule{}, nil
+	},
+}
+
+type languageDetectionModule struct{}
+
+func (l languageDetectionModule) GetStats() map[string]interface{} {
+	return nil
+}
+
+func (l languageDetectionModule) Register(router *module.Router) error {
+	router.HandleFunc("/detect", detectLanguage)
+	return nil
+}
+
+func (l languageDetectionModule) Close() {}
```

Comment:
worth adding a comment on why this doesn't do anything?

Reply:
Sure, basically, the `languageDetectionModule` doesn't do anything except route to the privileged language detection api. This API currently does not hold any resources over its lifetime, so there is no need to release any resources when the module is closed.